client/src/details/UrlBuilder.cpp

```diff
@@ -66,6 +66,31 @@ std::string UrlBuilder::GetUrl() const
     return urlPtr;
 }
 
+std::string UrlBuilder::GetPath() const
+{
+    CurlCharPtr path;
+    char* pathPtr = path.get();
+    THROW_IF_CURL_URL_SETUP_ERROR(curl_url_get(m_handle, CURLUPART_PATH, &pathPtr, 0 /*flags*/));
+    return pathPtr;
+}
+
+std::string UrlBuilder::GetQuery() const
+{
+    std::string ret;
+    CurlCharPtr query;
+    char* queryPtr = query.get();
+    const auto queryResult = curl_url_get(m_handle, CURLUPART_QUERY, &queryPtr, 0 /*flags*/);
+    if (queryResult == CURLUE_OK)
+    {
+        ret = queryPtr;

```

Comment:
Are you not using early-returns in this code base?
```
if (CURLUE_OK)
    return queryPtr;

if (CURLE_NO_QUERY)
    return {};

THROW_IF_CURL_URL_SETUP_ERROR(queryResult);
return {}; // keep compiler happy
```

Reply:
I find this easier to read and understand.

Reply:
Switched to a switch-case style with early returns since the pivot was the result